### PR TITLE
✂️✂️✂️  Part 1: Remove Sponsorship

### PR DESCRIPTION
**Pull request by Ridhwana**
## What type of PR is this? (check all applicable)

- [x] ✂️✂️✂️ 
- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
In this PR, I remove the sponsorship code from views, i8n, controllers, unused decorators, unused slack messengers and more.   

In a subsequent PR I'll remove the code from models, the database and do some more cleanup to the css and i8n files. The reason i separated it out like this is:
- I did not want this PR to be **even** bigger than it is
-  if I've missed anything in the views, it is less likely to break because the model and data are still in tact. 

## Related Tickets & Documents

- Related Issue #https://github.com/forem/forem/issues/18486
- Closes #

## QA Instructions, Screenshots, Recordings

This will be a bit difficult to test, but make sure that you have some sponsorships set up and live and then play around on the interface to make sure nothing breaks.

### UI accessibility concerns?

None

## Added/updated tests?

- [ ] Yes
- [x] No, and this is why: I've removed code
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://developers.forem.com) or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] This PR changes the Forem platform and our documentation needs to be
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
- [ ] I've updated the README or added inline documentation
- [ ] I've added an entry to
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [x] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [ ] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

File: app/controllers/tags_controller.rb [hunk 1]
```diff
@@ -80,7 +80,7 @@ def suggest
   private
 
   def tags
-    @tags ||= Tag.direct.includes(:sponsorship).limit(100)
```
Comment (Ridhwana) [inline app/controllers/tags_controller.rb:83]: We were showing tags that had sponsorship, but I've reverted the change and used the previous business logic which is to order the tags by their hotness score.

File: app/decorators/sponsorship_decorator.rb [hunk 2]
```diff
@@ -1,11 +0,0 @@
-class SponsorshipDecorator < ApplicationDecorator
-  def level_background_color
```
Comment (Ridhwana) [inline app/decorators/sponsorship_decorator.rb:2]: I checked for `level_background_color` in the code, but it seems like this decorator was not used anywhere, however we were testing decorators in general by using the SponsorshipDecrator. I remove these tests later on.

File: app/lib/constants/settings/general.rb [hunk 3]
```diff
@@ -99,10 +99,6 @@ def self.details
             description: I18n.t("lib.constants.settings.general.sidebar.description"),
             placeholder: I18n.t("lib.constants.settings.general.sidebar.placeholder")
           },
-          sponsor_headline: {
```
Comment (Ridhwana) [inline app/lib/constants/settings/general.rb:102]: This was being used on the admin in `app/views/admin/settings/forms/_sponsors.html.erb` which I've deleted.

File: app/models/admin_menu.rb [hunk 4]
```diff
@@ -46,7 +46,6 @@ class AdminMenu
     scope :advanced, "flashlight-line", [
       item(name: "broadcasts"),
       item(name: "response templates"),
-      item(name: "sponsorships"),
```
Comment (Ridhwana) [inline app/models/admin_menu.rb:49]: This removes sponsorship from the admin menu.

File: app/models/settings/general.rb [hunk 5]
```diff
@@ -93,9 +93,6 @@ class General < Base
     }
     setting :twitter_hashtag, type: :string
 
-    # Sponsors
-    setting :sponsor_headline, default: -> { I18n.t("models.settings.general.community_sponsors") }
```
Comment (Ridhwana) [inline app/models/settings/general.rb:97]: The headline was being used for the sponsorship widget in the sidebar which I've deleted.

File: app/services/slack/messengers/sponsorship.rb [hunk 6]
```diff
@@ -1,37 +0,0 @@
-module Slack
```
Comment (Ridhwana) [inline app/services/slack/messengers/sponsorship.rb:1]: I could not find a reference to this anywhere in the code, would appreciate a second pair of eyes on this please. Hence I've deleted it.

File: app/views/admin/settings/forms/_sponsors.html.erb [hunk 7]
```diff
@@ -1,20 +0,0 @@
-<%= form_for(Settings::General.new,
```
Comment (Ridhwana) [inline app/views/admin/settings/forms/_sponsors.html.erb:1]: The headline was being used for the sponsorship widget in the sidebar which I've deleted.

File: app/views/admin/sponsorships/edit.html.erb [hunk 8]
```diff
@@ -1,57 +0,0 @@
-<h2 class="crayons-title mb-6"><%= "Edit Sponsorship ##{@sponsorship.id}" %></h2>
```
Comment (Ridhwana) [inline app/views/admin/sponsorships/edit.html.erb:1]: CRUD views for sponsorship that I've deleted from the the admin.

File: app/views/admin/sponsorships/index.html.erb [hunk 9]
```diff
@@ -1,56 +0,0 @@
-<nav class="flex mb-4" aria-label="Sponsorships navigation">
```
Comment (Ridhwana) [inline app/views/admin/sponsorships/index.html.erb:1]: CRUD views for sponsorship that I've deleted from the the admin.

File: app/views/admin/sponsorships/new.html.erb [hunk 10]
```diff
@@ -1,66 +0,0 @@
-<h2 class="crayons-title mb-6">Make a new Sponsorship:</h2>
```
Comment (Ridhwana) [inline app/views/admin/sponsorships/new.html.erb:1]: CRUD views for sponsorship that I've deleted from the the admin.

File: app/views/articles/_sidebar.html.erb [hunk 11]
```diff
@@ -4,20 +4,7 @@
       <%= render partial: "shared/auth_widget", locals: { tracking_id: "ca_left_sidebar_home_page", source: "left_sidebar" } unless user_signed_in? %>
       <%= render partial: "layouts/main_nav", locals: { context: "sidebar" } %>
       <%= render "layouts/sidebar_tags" %>
-      <% @sponsorships = Sponsorship.gold.live.includes(:organization).order(featured_number: :asc) %>
       <%# the pattern .present?/.each has the advantage of issuing only a single SQL query to load objects in memory %>
-      <% if @sponsorships.present? %>
```
Comment (Ridhwana) [inline app/views/articles/_sidebar.html.erb:9]: This removes the sponsorship widget in the sidebar on the home page.

File: app/views/organizations/_sidebar_additional.html.erb [hunk 12]
```diff
@@ -1,11 +1,6 @@
 <div id="sidebar-wrapper-right" class="sidebar-wrapper sidebar-wrapper-right">
   <div class="sidebar-bg" id="sidebar-bg-right"></div>
   <div class="side-bar sidebar-additional showing" id="sidebar-additional">
-    <% @organization.sponsorships.live.find_each do |sponsorship| %>
```
Comment (Ridhwana) [inline app/views/organizations/_sidebar_additional.html.erb:4]: Removed the sponsorship badge on the organization page.

File: app/views/pages/_sponsor_image.html.erb [hunk 13]
```diff
@@ -1,2 +0,0 @@
-<img src="<%= optimized_image_url(sponsor.nav_image_url || sponsor.dark_nav_image_url || sponsor.profile_image_url) %>" width="250" alt="<%= sponsor.name %> logo" loading="lazy" class="partner-image-light-mode">
```
Comment (Ridhwana) [inline app/views/pages/_sponsor_image.html.erb:1]: Deleted as it was referenced in app/views/articles/_single_sponsor.html.erb which was deleted.

File: app/views/stories/tagged_articles/_sidebar.html.erb [hunk 14]
```diff
@@ -29,16 +29,6 @@
           </div>
         <% end %>
       <% end %>
-      <% if @tag.sponsorship && @tag.sponsorship.status == "live" && @tag.sponsorship.expires_at > Time.current %>
```
Comment (Ridhwana) [inline app/views/stories/tagged_articles/_sidebar.html.erb:32]: Removed widget for tag sponsorship.

File: app/views/tags/index.html.erb [hunk 15]
```diff
@@ -57,9 +57,6 @@
           <% end %>
           <p class="mb-3 fs-s color-base-60">
             <%= t("views.tags.published", count: tag.taggings_count) %>
-            <% if tag.sponsorship && tag.sponsorship.status == "live" && tag.sponsorship.expires_at > Time.current %>
```
Comment (Ridhwana) [inline app/views/tags/index.html.erb:60]: Removed widget for tag sponsorship.

File: config/routes.rb [hunk 16]
```diff
@@ -220,7 +220,6 @@
     get "/checkin", to: "pages#checkin"
     get "/💸", to: redirect("t/hiring")
     get "/survey", to: redirect("https://dev.to/ben/final-thoughts-on-the-state-of-the-web-survey-44nn")
-    get "/sponsors", to: "pages#sponsors"
```
Comment (Ridhwana) [inline config/routes.rb:223]: Removed pages sponsorship.

File: lib/tasks/add_navigation_links.rake [hunk 17]
```diff
@@ -1,7 +1,7 @@
 # rubocop:disable Metrics/BlockLength
 namespace :navigation_links do
   def image_path(*paths)
-    File.read(Rails.root.join("app/assets/images/#{paths.join('/')}")).freeze
```
Comment (Ridhwana) [inline lib/tasks/add_navigation_links.rake:4]: This was rubocop.

File: lib/tasks/add_navigation_links.rake [hunk 18]
```diff
@@ -192,14 +191,6 @@ namespace :navigation_links do
       position: 6,
       section: :default,
     )
-    NavigationLink.create_or_update_by_identity(
```
Comment (Ridhwana) [inline lib/tasks/add_navigation_links.rake:195]: Remove sponsors link on the user facing side.

File: spec/decorators/application_decorator_spec.rb [hunk 19]
```diff
@@ -35,22 +35,22 @@
...
+      create(:article, approved: true)
     end
 
     it "receives an ActiveRecord relation and returns an array of decorated records" do
-      relation = Sponsorship.gold
+      relation = Article.approved
 
       decorated_collection = described_class.decorate_collection(relation)
-      expect(decorated_collection.map(&:class)).to eq([SponsorshipDecorator])
+      expect(decorated_collection.map(&:class)).to eq([ArticleDecorator])
```
Comment (Ridhwana) [inline spec/decorators/application_decorator_spec.rb:45]: We changed the example to use the article decorator instead of sponsorship decorator.

**Review by jaw6 — APPROVED** [review]
🎉 Woooo

**Review by fdocr — APPROVED** [review]
✂️ ✂️ ✂️

**Pull request merged**